src/animation/Interpolant.js

```diff
@@ -0,0 +1,220 @@
+/**
+ *
+ * Abstract base class for interpolants over timed keyframe values.
+ * It handles seeking of the interval and boundary cases. Concrete
+ * subclasses then implement the actual intepolation by filling in
+ * the Template Methods.
+ *
+ *
+ * @author tschw
+ */
+
+THREE.Interpolant = function( times, values, stride, result ) {
+
+	this.times = times;
+	this.values = values;
+	this.stride = stride;
+
+	this.result = result;
+
+	this.cachedIndex = 0;
+
+};
+
+THREE.Interpolant.prototype = {
+
+	constructor: THREE.Intepolant,
+
+	getAt: function( time ) {
```

Comment:
I'd suggest unit tests or something similar to ensure correctness of this function.  It is involved and there area  lot of edge cases that we need to ensure stay covered.  It may be correct now, and it is important to keep it that way.